mistralrs-core/src/vision_models/llava/llava_inputs_processor.rs

```diff
@@ -0,0 +1,328 @@
+#![allow(clippy::cast_possible_truncation, clippy::cast_precision_loss)]
+use std::any::Any;
+use std::sync::Arc;
+
+use candle_core::Result;
+use candle_core::{DType, Device, Tensor};
+use image::GenericImageView;
+use image::Rgb;
+use itertools::Itertools;
+use regex_automata::meta::Regex;
+use tokenizers::Tokenizer;
+
+use super::llava15::LLaVAVisionSpecificArgs;
+use super::utils::{expand2square, LLaVAImageProcessor};
+use crate::pipeline::text_models_inputs_processor::{get_completion_input, get_prompt_input};
+use crate::pipeline::{
+    text_models_inputs_processor, InputsProcessor, InputsProcessorType, MessagesAction, Processor,
+};
+use crate::sequence::Sequence;
+use crate::vision_models::image_processor::{self, ImagePreProcessor, PreprocessedImages};
+use crate::vision_models::llava::config::Config as LLaVAConfig;
+use crate::vision_models::preprocessor_config::{PreProcessorConfig, ToFilter};
+use crate::vision_models::{preprocessor_config, ModelInputs};
+
+pub struct LLaVAProcessor {
+    inputs_processor: Arc<LLaVAInputProcessor>,
+}
+
+impl Processor for LLaVAProcessor {
+    fn inputs_processor(&self) -> Arc<dyn InputsProcessor> {
+        self.inputs_processor.clone()
+    }
+    fn get_special_tokens(&self) -> &[&'static str] {
+        &[]
+    }
+    fn template_action(&self) -> MessagesAction {
+        MessagesAction::FlattenOnlyText
+    }
+}
+
+impl LLaVAProcessor {
+    pub fn new(config: &str) -> Self {
+        let model_config =
+            serde_json::from_str::<LLaVAConfig>(config).expect("Failed to parse model config.");
+        let image_tag_splitter =
+            Regex::new(r"<\|image_\d+\|>").expect("Failed to compile split regex.");
```

Comment:
Is this the right image token? From the HF model card I thought it was `<image>`.